### ISSUE TITLE
openbr Segmentation Fault

### ISSUE THREAD

**Issue by cherbin** [biometrics/openbr#233]
I've just installed openbr with all the dependencies. There seems to be no problem with the install 

I run "br" command as follows: 

br -algorithm FaceRecognition -compare face1.jpg face2.jpg
Segmentation fault (core dumped)

What possibly could be the issue ? look below for the test program too.

ALSO
I went to run the test and get the following results : 

Test project /home/ubuntu/testing/Install-OpenCV-master/Ubuntu/openbr/build
    Start 1: br_initialize
1/9 Test #1: br_initialize ......................**_Exception: SegFault  0.15 se                                                                         c
    Start 2: br_objects
2/9 Test #2: br_objects .........................**_Exception: SegFault  0.14 se                                                                         c
    Start 3: br_draw_face_detection
3/9 Test #3: br_draw_face_detection .............**_Exception: SegFault  0.15 se                                                                         c
    Start 4: age_estimation_test
4/9 Test #4: age_estimation_test ................**_Exception: SegFault  0.15 se                                                                         c
    Start 5: face_recognition_search_test
5/9 Test #5: face_recognition_search_test .......**_Exception: SegFault  0.14 se                                                                         c
    Start 6: face_recognition_evaluation_test
6/9 Test #6: face_recognition_evaluation_test ...**_Exception: SegFault  0.15 se                                                                         c
    Start 7: face_recognition_train_test
7/9 Test #7: face_recognition_train_test ........**_Exception: SegFault  0.15 se                                                                         c
    Start 8: gender_estimation_test
8/9 Test #8: gender_estimation_test .............**_Exception: SegFault  0.15 se                                                                         c
    Start 9: face_recognition_test
9/9 Test #9: face_recognition_test ..............***Exception: SegFault  0.15 se                                                                         c

0% tests passed, 9 tests failed out of 9

Total Test time (real) =   1.33 sec

The following tests FAILED:
          1 - br_initialize (SEGFAULT)
          2 - br_objects (SEGFAULT)
          3 - br_draw_face_detection (SEGFAULT)
          4 - age_estimation_test (SEGFAULT)
          5 - face_recognition_search_test (SEGFAULT)
          6 - face_recognition_evaluation_test (SEGFAULT)
          7 - face_recognition_train_test (SEGFAULT)
          8 - gender_estimation_test (SEGFAULT)
          9 - face_recognition_test (SEGFAULT)
Errors while running CTest

**Comment by mikaelwestoo** [comment]
I have the exact same problem on a fresh Amazon-Ubuntu-server,  when will this be solved?

**Comment by kheera** [comment]
Are you running with option: -useGui 0 ?
Something like this:  br -useGui 0 -algorithm FaceRecognition -enrollAL -enroll faces/ face_data.csv

**Comment by mikaelwestoo** [comment]
root@ip-xxxxxxx:~/files/br_master/openbr/build/app/br# ./br -useGui 0
Segmentation fault (core dumped)

**Comment by kheera** [comment]
Not really sure... running through gdb might help figure out what the problem is.

**Comment by mikaelwestoo** [comment]
What did you downgrade in order to get it to work?  You wrote "made sure only qt5 was installed, downgraded to v0.4.1 to" ...    I can not find (via git) openbr or qt on version 0.4.1...

**Comment by kheera** [comment]
Changed to openbr-0.4.1 to get past a compile error.

**Comment by mikaelwestoo** [comment]
(gdb) run
Starting program: /mnt2/milko/files/br_master/openbr/build/app/br/br
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

Program received signal SIGSEGV, Segmentation fault.
0x00007fffed062e1f in ?? () from /usr/lib/x86_64-linux-gnu/libQtTest.so.4
(gdb)

**Comment by mikaelwestoo** [comment]
When I try running qrcreator by itself I get.. 


root@xxxx:~/files/br_master/openbr/build/app/br# qtcreator -useGui 0
QXcbConnection: Could not connect to display
Aborted (core dumped)

root@xxxx:~/files/br_master/openbr/build/app/br# qtcreator
QXcbConnection: Could not connect to display
Aborted (core dumped)

**Comment by mikaelwestoo** [comment]
(gdb) run
Starting program: /usr/bin/qtcreator
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
QXcbConnection: Could not connect to display

Program received signal SIGABRT, Aborted.
0x00007ffff5878cc9 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
56      ../nptl/sysdeps/unix/sysv/linux/raise.c: No such file or directory.

**Comment by kheera** [comment]
Your first gdb post indicates qt4 is being used. I'd start fresh again, remove all versions of qt4 from the system, install qt5, then install openbr 0.4.1.

**Comment by mikaelwestoo** [comment]
OK, I will report on my progress.. on what I did to make it work. Thanks for great support!

**Comment by caotto** [comment]
As a side note, QtCreator is a GUI application, so it is expected that it
will not work on your server.

**Comment by mikaelwestoo** [comment]
So, after 10 different compiles and apt-get´s I now get 

root@ip-1xxxxxxx:~/files/br_master/openbr/build/app/br# ./br
libdc1394 error: Failed to initialize libdc1394
OpenBR 0.6.0 Copyright (c) 2013 OpenBiometrics. All rights reserved.
Try running 'br -help'

--- So this is what I did, from what I remember and documented.----  

First what I did is that I tried to uninstall any type of QT4 on the Ubuntu 14.*

Run these commands..
apt-get remove qt4-linguist-tools
apt-get remove qt4-qmake
apt-get autoremove

Then I found this, do not know if it is necessary, but I installed it anyway:
apt-get install qttools5-dev-tools

I tried to compile new QT5, but since the server is an Amazon server with no gui, that was a no go.

What I noticed is that the problem lied in the compilation of the "cv2" ( opencv ). So I went back to that library and remove the build and ran 

cmake -DCMAKE_BUILD_TYPE=Release ..

My main problem was that got "QT" = NO

--   GUI:
--     QT:                          NO
--     GTK+ 2.x:                    YES (ver 2.24.23)
--     GThread :                    YES (ver 2.40.2)


But since it was only under "GUI", I skipped that requirement and went ahead and compiled opencv anyways.. And opencv did compile successfull.

I went ahead and cloned the "master"-version of BR..  and it succeeded! :)

So, my guess is that you need to remove ( as it was mentioned in this thread before ) QT4..

I used these commands ( before compiling opencv )

I have also noticed that my server had about ~50 libqt5*  ( apt-get install libqt5* ) packages installed.. So it is pretty fair to say that QT5 is installed on the server.

Anyways, it worked for me:


TL:DR:

1. Download new "master"-branch-sources
2. Remove any qt4-lib on the server
3. I ran these commands:
apt-get remove qt4-linguist-tools
apt-get remove qt4-qmake
apt-get autoremove

4. Then I found this, do not know if it is necessary, but I installed it anyway:
apt-get install qttools5-dev-tools

5. Compile opencv ( cv2 )
6. Compile openbr
7. Working.

**Comment by sklum** [comment]
This seems to be taken care of.
